src_c/polygon.c

```diff
@@ -449,9 +517,137 @@ pg_polygon_collidepoint(pgPolygonObject *self, PyObject *const *args,
     return PyBool_FromLong(pgCollision_PolygonPoint(&self->polygon, px, py));
 }
 
+static void
+_pg_rotate_polygon_helper(double *vertices, Py_ssize_t verts_num, double angle,
+                          double c_x, double c_y)
+{
+    Py_ssize_t i2;
+    /*convert the angle to radians*/
+    double angle_rad = angle * M_PI / 180.0;
+
+    if (fmod(angle_rad, M_PI_2) != 0.0) {
+        /* handle the general angle case that's not 90, 180 or 270 degrees */
+        double cos_a = cos(angle_rad) - 1;
+        double sin_a = sin(angle_rad);
+
+        for (i2 = 0; i2 < verts_num * 2; i2 += 2) {
+            double dx = vertices[i2] - c_x;
+            double dy = vertices[i2 + 1] - c_y;
+            vertices[i2] += dx * cos_a - dy * sin_a;
+            vertices[i2 + 1] += dx * sin_a + dy * cos_a;
+        }
+        return;
+    }
+
+    /*Ensure angle is between 0 and two pi*/
+    angle_rad = fmod(angle_rad, M_TWOPI);
+    if (angle_rad < 0) {
+        angle_rad += M_TWOPI;
+    }
+
+    double v1, v2;
+    /*special-cases rotation by 90, 180 and 270 degrees*/
+    switch ((int)(angle_rad / M_PI_2)) {
+        case 1:
```

Comment:
I actually never thought of this. Good job.

Reply:
actually it's not my idea, it's something i saw in the pygame.math module in the vector rotate methods ahah.

Reply:
it's a bit more complex tho obviously